cluster-cm/src/main/java/com/sequenceiq/cloudbreak/cm/ClouderaManagerClusterStatusService.java

```diff
@@ -282,6 +297,26 @@ public ExtendedHostStatuses getExtendedHostStatuses(Optional<String> runtimeVers
         return new ExtendedHostStatuses(hostStates);
     }
 
+    @Override
+    public DetailedHostStatuses getDetailedHostStatuses() {
+        List<ApiHost> apiHosts = getHostsFromCM();
+        Map<HostName, Optional<DetailedHostHealthCheck>> hostsHealth = apiHosts.stream().collect(Collectors.toMap(
+                apiHost -> hostName(apiHost.getHostname()),
+                ClouderaManagerClusterStatusService::getDetailedHostHealthCheck));
+
+        Map<HostName, Optional<DetailedServicesHealthCheck>> servicesHealth = apiHosts.stream().collect(Collectors.toMap(
+                apiHost -> hostName(apiHost.getHostname()),
+                ClouderaManagerClusterStatusService::getDetailedServicesHealthCheck));
+
+        Map<HostName, Optional<DetailedCertHealthCheck>>  certHealth = apiHosts.stream().collect(Collectors.toMap(
+                apiHost -> hostName(apiHost.getHostname()),
+                ClouderaManagerClusterStatusService::getDetailedCertificateCheck));
+
+        LOGGER.debug("Creating 'DetailedHostStatuses' with {}", hostsHealth);
+
+        return new DetailedHostStatuses(hostsHealth, servicesHealth, certHealth);
```

Comment:
the old logic skips the services check if runtime is older than X, that logic is missing here, don't we need that?

Reply:
I tested this with the pre 7.2.12 runtime [7.2.7] (which is the exact runtime check that happens in `ExtendedHostStatuses`), and this API exists, (and was introduced in CM API v40) however, it looks like a runtime check was added via [CB-13723](https://jira.cloudera.com/browse/CB-13723) due to [OPSAPS-61389](https://jira.cloudera.com/browse/OPSAPS-61389). The runtime check was bypassed here because of the existence of this API in pre 7.2.12 clusters

Reply:
makes sense, thx